src/overlays/actors/ovl_En_Rd/z_en_rd.c

```diff
@@ -173,83 +205,97 @@ void EnRd_Destroy(Actor* thisx, GlobalContext* globalCtx) {
     if (gSaveContext.sunsSongState != SUNSSONG_INACTIVE) {
         gSaveContext.sunsSongState = SUNSSONG_INACTIVE;
     }
+
     Collider_DestroyCylinder(globalCtx, &this->collider);
 }
 
-void func_80AE2630(GlobalContext* globalCtx, Actor* thisx, s32 arg2) {
-    Actor* enemyIt = globalCtx->actorCtx.actorLists[ACTORCAT_ENEMY].head;
-
-    while (enemyIt != NULL) {
-        if ((enemyIt->id != ACTOR_EN_RD) || (enemyIt == thisx) || (enemyIt->params < 0)) {
-            enemyIt = enemyIt->next;
+/**
+ * This function does two things depending on whether shouldMourn is true or false.
+ * - If shouldMourn is true, this function sets thisx to be the parent for all Redeads
+ *   in the area that are capable of mourning. This is used right when thisx first
+ *   dies to make the other Redeads mourn it.
+ * - If shouldMourn is false, this function nulls out the parent for all Redeads in the
+ *   are whose parents is thisx. This is used when thisx is fading away to make the
```

Comment:
just a typo I didn't notice before
```suggestion
 *   area whose parent is thisx. This is used when thisx is fading away to make the
```

Reply:
Good catch, though I think the grammatically correct thing to say is "all Redead*s* in the area whose parent*s* *are* thisx", since we're referring to multiple redeads. It's spread across two lines so it's difficult to see, and the "are" that's there right now is just totally wrong.

Reply:
Sure, I was indeed hesitating